op-chain-ops/cmd/celo-migrate/db.go

```diff
@@ -68,25 +68,6 @@ func blockReceiptsKey(number uint64, hash common.Hash) []byte {
 	return append(append(blockReceiptsPrefix, encodeBlockNumber(number)...), hash.Bytes()...)
 }
 
-// Opens a database with access to AncientsDb
-func openDB(chaindataPath string, readOnly bool) (ethdb.Database, error) {
-	// Will throw an error if the chaindataPath does not exist
-	if _, err := os.Stat(chaindataPath); err != nil {
-		return nil, err
-	}
-
-	kvs, err := leveldb.New(chaindataPath, 1024, 60, "", readOnly)
-	if err != nil {
-		return nil, fmt.Errorf("failed to open leveldb: %w", err)
-	}
-	db, err := rawdb.NewDatabaseWithFreezer(kvs, filepath.Join(chaindataPath, "ancient"), "", false)
```

Comment:
For me the problem appears to happen because `NewDatabaseWithFreezer` is called with readonly == false here 🤦‍♂️. I have no idea how I managed to overlook this or why it wasn't causing problems before. 

When I pass in the readonly boolean here I can't reproduce the issue even without other changes. That being said it might still make sense to add the other changes you've made here and get rid of the openDB function entirely.